src/BotiumBotmockExporter.ts

```diff
@@ -0,0 +1,443 @@
+                    asserters: [
+                        {
+                            name: 'INTENT',
+                            args: [nextMessage.intent.label]
+                        }
+                    ]
+                })
+            }
+            this.#getConversationsRecursive(
+                nextMessage.message_id,
+                messages,
+                results,
+                botiumConvoStepsCloned,
+                processedMessageIdCloned,
+                logsCloned
+            )
+        });
+
+        return results
+    };
+
+    #txtConvoTransformation = (
+        resources: Resources,
+        messages: Map<string, Botmock.Message>,
+        intents: Map<string, Botmock.Intent>,
+        rootMessages: string[]
+    ) => {
+        const extractedConversations: ResultEntry[] = [];
+        for (const rootMessageId of rootMessages) {
+            this.#getConversationsRecursive(rootMessageId, messages, extractedConversations);
+        }
+
+        const driver = new BotDriver();
+        const compiler = driver.BuildCompiler()
+
+        return extractedConversations.map(entry => {
+            const {
+                botiumConvoSteps,
+                logs
+            } = entry
+
+            const decisions = logs.filter(log => log.connectionBranchCount && log.connectionBranchCount > 1).map(log => (log.connectionBranchIndex || 0) + 1)
+
+            const convoGeneratedName = sanitize(resources.project.name) + (decisions.length ? `_${decisions.join('_')}` : '')
+            const convoGeneratedDescription = logs.flatMap(log => {
+                const result = []
+                if (log.type === 'node') {
+                    result.push(`Node "${log.nodeMessageId}": ${log.nodeName} (${log.nodeMessageType})`)
+                    if (log.nodeTerminatedReason) {
+                        result.push(log.nodeTerminatedReason)
+                    }
+                    return result
+                } else {
+                    result.push(`Connection "${(log.connectionBranchIndex || 0) + 1}/${(log.connectionBranchCount || 0)}":${' '.repeat(28) + JSON.stringify({
+                        intent: log.connectionBranchingReason?.intent,
+                        conditional: log.connectionBranchingReason?.conditional
+                    })}`)
+                    if (log.connectionTerminatedReason) {
+                        result.push(log.connectionTerminatedReason)
+                    }
+
+                    return result
+                }
+            }).join('\n')
+            const convo: Botium.Convo = {
+                header: {
+                    name: convoGeneratedName,
+                    description: convoGeneratedDescription
+                },
+                conversation: botiumConvoSteps
+            }
+
+            const data = compiler.Decompile([convo], SCRIPTING_FORMAT_TXT, SCRIPTING_TYPE_CONVO)
+            var fs = require('fs');
```

Comment:
I think fs could be imported globally in the file's header:
`import { writeFileSync } from 'fs'`
You need to install node @types/node first: npm install @types/node --save-dev